test/livebook/live_markdown/markdown_helpers_test.exs

```diff
@@ -24,16 +34,56 @@ defmodule Livebook.LiveMarkdown.MarkdownHelpersTest do
       assert markdown == reformat(markdown)
     end
 
+    test "inline code with a line break" do
+      markdown = "The Game, `Mrs\nHudson`, is on!"
+      assert markdown == reformat(markdown)
+    end
+
+    test "inline code with extra spaces" do
+      markdown = "The Game, `Mrs Huds  on`, is on!"
+      assert markdown == reformat(markdown)
+    end
+
     test "combined" do
       markdown = "The Game, ~~***`Mrs Hudson`***~~, is on!"
       assert markdown == reformat(markdown)
     end
 
+    test "inline math" do
+      markdown = "The Game, $Mrs_{ij} Hudson_{ij}$, is on!"
+
+      assert markdown == reformat(markdown)
+    end
+
+    test "alternate brackets for inline math" do
+      markdown = "The Game, \\(Mrs_{ij} Hudson_{ij}\\), is on!"
+
+      assert markdown == reformat(markdown)
+    end
+
     test "link" do
       markdown = "The Game, [Mrs Hudson](https://youtu.be/M-KqaO1oH2E), is on!"
       assert markdown == reformat(markdown)
     end
 
+    test "link with IAL" do
+      markdown = "[link](url){: .classy}"
+
+      assert markdown == reformat(markdown)
+    end
+
+    test "link followed by escaped IAL" do
+      markdown = "[link](url)\\{: .classy}"
+
+      assert markdown == reformat(markdown)
```

Comment:
The first time this Markdown is formatted, the `\` disappears. The second time, the no-longer-escaped IAL annotation does too. :scream: